connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/DeadLetterQueueReporter.java

```diff
@@ -121,20 +121,22 @@ public static DeadLetterQueueReporter createAndSetup(Map<String, Object> adminPr
     /**
      * Write the raw records into a Kafka topic and return the producer future.
      *
-     * @param context processing context containing the raw record at {@link ProcessingContext#consumerRecord()}.
+     * @param context processing context containing the raw record at {@link ProcessingContext#original()}.
      * @return the future associated with the writing of this record; never null
      */
-    public Future<RecordMetadata> report(ProcessingContext context) {
+    @SuppressWarnings("unchecked")
+    public Future<RecordMetadata> report(ProcessingContext<?> context) {
         if (dlqTopicName.isEmpty()) {
             return CompletableFuture.completedFuture(null);
         }
         errorHandlingMetrics.recordDeadLetterQueueProduceRequest();
 
-        ConsumerRecord<byte[], byte[]> originalMessage = context.consumerRecord();
-        if (originalMessage == null) {
+        if (!(context.original() instanceof ConsumerRecord)) {
             errorHandlingMetrics.recordDeadLetterQueueProduceFailed();
             return CompletableFuture.completedFuture(null);
         }
+        ProcessingContext<ConsumerRecord<byte[], byte[]>> sinkContext = (ProcessingContext<ConsumerRecord<byte[], byte[]>>) context;
```

Comment:
Just to be safe, should this be `ProcessingContext<? extends ConsumerRecord<byte[], byte[]>>`?

Reply:
Yes, but ConsumerRecord has no subclasses, and we never create a ProcessingContext for anything other than SourceRecord and ConsumerRecord<byte[], byte[]>.

Reply:
Hmmm, I was wondering if a consumer interceptor could potentially return a subclass of `ConsumerRecord` but this doesn't seem possible at the moment. Fine as-is, then 👍